src/ast/analysis/Type.cpp

```diff
@@ -503,6 +510,256 @@ void TypeAnalysis::run(const TranslationUnit& translationUnit) {
+        for (std::size_t i = 0; i < arguments.size(); ++i) {
+            TypeAttribute argType = typeAnalysis.getFunctorParamTypeAttribute(fun, i);
+            auto& ty = typeEnv.getConstantType(argType);
+            branchOnArgument(arguments[i], ty);
+            if (i + 1 < arguments.size()) {
+                os << ",";
+            }
+        }
+        os << ")";
+    }
+}
+
+void TypeAnnotationPrinter::print_(type_identity<UserDefinedFunctor>, const UserDefinedFunctor& fun) {
+    auto arguments = fun.getArguments();
+    os << "@" << fun.getName() << "(";
+    for (std::size_t i = 0; i < arguments.size(); ++i) {
+        TypeAttribute argType = typeAnalysis.getFunctorParamTypeAttribute(fun, i);
+        auto& ty = typeEnv.getConstantType(argType);
+        branchOnArgument(arguments[i], ty);
+        if (i + 1 < arguments.size()) {
+            os << ",";
+        }
+    }
+    os << ")";
+}
+
+void TypeAnnotationPrinter::print_(type_identity<Counter>, [[maybe_unused]] const Counter& counter) {
+    os << "$∈{number}";
+}
+
+void TypeAnnotationPrinter::print_(type_identity<TypeCast>, const ast::TypeCast& typeCast) {
+    os << "as(";
+    auto& ty = typeEnv.getType(typeCast.getType());
+    branchOnArgument(typeCast.getValue(), ty);
+    os << "," << ty.getName() << ")";
+}
+
+void TypeAnnotationPrinter::print_(
+        type_identity<RecordInit>, const RecordInit& record, const RecordType& type) {
+    auto arguments = record.getArguments();
+    auto& ftypes = type.getFields();
+    os << "[";
+    for (std::size_t i = 0; i < arguments.size(); ++i) {
+        branchOnArgument(arguments[i], *ftypes[i]);
+        if (isA<RecordInit>(arguments[i]) || isA<UnnamedVariable>(arguments[i])) {
+            os << "∈{" << (*ftypes[i]).getName() << "}";
+        }
+        if (i + 1 < arguments.size()) {
+            os << ",";
+        }
+    }
+    os << "]";
+}
+
+void TypeAnnotationPrinter::print_(type_identity<BranchInit>, const BranchInit& adt) {
+    auto* correspondingType = sumTypesBranches.getType(adt.getBranchName());
+
+    assert(correspondingType != nullptr);
+    assert(isA<AlgebraicDataType>(correspondingType));
+
+    auto branchTypes = as<AlgebraicDataType>(correspondingType)->getBranchTypes(adt.getBranchName());
+    auto branchArgs = adt.getArguments();
+
+    assert(branchTypes.size() == branchArgs.size());
+
+    os << "$" << adt.getBranchName() << "(";
+    for (std::size_t i = 0; i < branchArgs.size(); i++) {
+        auto arg = branchArgs[i];
+        auto argTy = branchTypes[i];
+        branchOnArgument(arg, *argTy);
+        if (isA<RecordInit>(arg) || isA<UnnamedVariable>(arg)) {
+            os << "∈{" << argTy->getName() << "}";
+        }
+        if (i + 1 < branchArgs.size()) {
+            os << ", ";
+        }
+    }
+    os << ")";
+}
+
+void TypeAnnotationPrinter::print_(type_identity<Aggregator>, [[maybe_unused]] const Aggregator& agg) {
+    // TODO
+}
+
+// Note: branchOnArgument do the better job.
```

Comment:
Please can you remove the Note and TODO above.

Reply:
Instead of deleting those unfinished parts, I implemented functions for supporting aggregator. How is it?